src/olympia/git/utils.py

```diff
@@ -717,3 +721,51 @@ def _render_patch(self, patch, commit, parent, pathspec=None):
         }
 
         return entry
+
+
+def skip_git_extraction(version):
+    return (
+        version.addon.type != amo.ADDON_EXTENSION or
+        not version.all_files[0].is_webextension
+    )
+
+
+def create_git_extraction_entry(version):
+    if skip_git_extraction(version):
+        log.debug('Skipping git extraction of add-on "{}": not a '
+                  'web-extension.'.format(version.addon.id))
+        return
+
+    log.info('Adding add-on "{}" to the git extraction '
+             'queue.'.format(version.addon.id))
+    GitExtractionEntry.objects.create(addon=version.addon)
+
+
+def extract_version_to_git(version_id):
+    """Extract a `Version` into our git storage backend."""
+    # We extract deleted or disabled versions as well so we need to make sure
+    # we can access them.
+    version = Version.unfiltered.get(pk=version_id)
+
+    if skip_git_extraction(version):
+        # We log a warning message because this should not happen (as the CRON
+        # task should no select non-webextension versions).
+        log.warning('Skipping git extraction of add-on "{}": not a '
+                    'web-extension.'.format(version.addon.id))
+        return
+
+    log.info('Extracting {version_id} into git backend'.format(
+        version_id=version_id))
+
+    try:
+        with statsd.timer('git.extraction.version'):
+            repo = AddonGitRepository.extract_and_commit_from_version(
+                version=version
+            )
+        statsd.incr('git.extraction.version.success')
+    except Exception as exc:
+        statsd.incr('git.extraction.version.failure')
```

Comment:
I see you have a test that checks the `success` version of `incr`, but I don't see a test for this `failure` case. Should there be one?

Reply:
For better coverage, yes probably